multi-agent/adapters/inbound/flask/decorators.py

```
@@ -221,36 +222,101 @@ def with_identity(f: Callable) -> Callable:
+    ``current_app.container.admin_config_reader`` (centralized admin
+    config managed via the admin panel).  Falls back to the static
+    ``admin_allowed_users`` Flask config.
+
+    All admin-gated surfaces (built-in resources, OAuth client config,
+    templates, statistics) share this single gate intentionally — there
+    is one admin concept across the MAS service.
+
     Does NOT inject ``identity`` into the wrapped function's kwargs.
     """
     @wraps(f)
     def decorated_function(*args, **kwargs):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `decorated_function`

(ANN202)